### PR TITLE
Add a new endpoint to refresh spotify tokens

### PR DESCRIPTION
**Pull request by paramsingh**
# Summary

* This is a…
    * ( ) Bug fix
    * (x) Feature addition
    * ( ) Refactoring
    * ( ) Minor / simple change (like a typo)
    * ( ) Other

# Problem

* JIRA ticket (_optional_): 

Tokens expired, leading to errors in the frontend.

# Solution

I added an endpoint to which you can send GET requests to get a new token if needed.

Also, I improved the error handling to raise API specific errors which return json instead of html.

### CODE REVIEW

File: listenbrainz/webserver/views/api.py [hunk 1]
```diff
@@ -290,29 +290,36 @@ def validate_token():
             'message': 'Token valid.'
         })
 
+
+@api_bp.route('/refresh-spotify-token', methods=['POST'])
+@crossdomain(headers='Authorization, Content-Type')
+def refresh_spotify_token():
```
Comment (alastair) [inline listenbrainz/webserver/views/api.py]: What's this method used for?

File: listenbrainz/webserver/views/profile.py [hunk 2]
```diff
@@ -258,3 +261,23 @@ def connect_spotify_callback():
         flash.warn('Unable to authenticate with Spotify (error {})'.format(e.args[0]))
 
     return redirect(url_for('profile.connect_spotify'))
+
+
+@profile_bp.route('/refresh-spotify-token', methods=['GET'])
```
Comment (alastair) [inline listenbrainz/webserver/views/profile.py]: This feels like it should be POST, not GET

File: listenbrainz/webserver/views/profile.py [hunk 3]
```diff
@@ -258,3 +261,23 @@ def connect_spotify_callback():
...
+@crossdomain()
+@api_login_required
+def refresh_spotify_token():
+    spotify_user = spotify.get_user(current_user.id)
+    if not spotify_user:
+        raise APINotFound("User has not authenticated to Spotify")
```
Comment (alastair) [inline listenbrainz/webserver/views/profile.py]: I don't know if 404 is a good error here. The page still exists, it just can't be run. 400 doesn't seem right either:
> the server cannot or will not process the request due to something that is perceived to be a client error

Maybe 409?

File: listenbrainz/webserver/views/profile.py (hunk 2 above)
Comment (MonkeyDo) [reply listenbrainz/webserver/views/profile.py]: It's defined as such for the api:
https://github.com/metabrainz/listenbrainz-server/blob/f61e811ddee8911990dcfdf9893115be0646c816/listenbrainz/webserver/views/api.py#L294

File: listenbrainz/webserver/views/profile.py [hunk 4]
```diff
@@ -258,3 +261,23 @@ def connect_spotify_callback():
...
+    if spotify_user.token_expired:
+        try:
+            spotify_user = spotify.refresh_user_token(spotify_user)
+        except spotify.SpotifyAPIError:
+            raise APIServiceUnvailable("Cannot refresh Spotify token right now")
```
Comment (alastair) [inline listenbrainz/webserver/views/profile.py]: Unavailable

**Pull request merged**